docs/4.4/api-reference.md

```diff
@@ -11,57 +11,287 @@ Teleport is currently working on documenting our API.
+  },
+})
+if err != nil {
+  return err
+}
+
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+**Retrieve role**
+
+```go
+role, err := client.GetRole("auditor")
+if err != nil {
+  return err
+}
+```
+
+**Update role**
+
+```go
+// update the auditor role's read rule to not provide access to secrets
+role.SetRules(services.Allow, []services.Rule{
+	services.NewRule(services.KindSession, services.ReadNoSecrets()),
+})
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+**Delete role**
+
+```go
+if err := client.DeleteRole(ctx, "auditor"); err != nil {
+  return err
+}
+```
+
+## Tokens
+
+[Tokens](admin-guide.md#adding-nodes-to-the-cluster) are used to add nodes to clusters, and add remote clusters to [trusted clusters](admin-guide.md#trusted-clusters).
+
+**Create token**
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  // You can provide 'Token' for a non-random tokenString
+  Roles: teleport.Roles{teleport.RoleProxy},
+  TTL:   time.Hour,
+})
+if err != nil {
+  return err
+}
+```
+
+**Retrieve token**
 
-## Audit Events API
-Coming Soon
+```go
+token, err := client.GetToken(tokenString)
+if err != nil {
+  return err
+}
+```
+
+**Update token**
+
+```go
+// updates the token to be a proxy token
+token.SetRoles(teleport.Roles{teleport.RoleProxy})
+if err := client.UpsertToken(token); err != nil {
+  return err
+}
+```
+
+**Delete token**
+
+```go
+if err := client.DeleteToken(tokenString); err != nil {
+  return err
+}
+```
+
+## Cluster Labels
+
+The root cluster in a [trusted cluster](admin-guide.md#adding-nodes-to-the-cluster) can add/update cluster labels on remote clusters.
+
+**Create a remote cluster token with Labels**
+
+Remote clusters will inherit the labels of the [join token](trustedclusters.md#join-tokens) used to add them to the trusted cluster. This is the preferred method of managing cluster tokens.
+
+Follow the trusted cluster join token [docs](trustedclusters.md#join-tokens) to use this token to create a remote cluster using `tctl`.
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  // You can provide 'Token' for a static token name
+  Roles: teleport.Roles{teleport.RoleTrustedCluster},
+  TTL:   time.Hour,
+  Labels: map[string]string{
+    "env": "staging",
+  },
+})
+```
+
+**Update remote cluster's labels**
+
+You can also update a remote cluster's labels from the root cluster using the auth api if necessary.
+
+```go
+rc, err := services.GetRemoteCluster("remoteClusterName")
+if err != nil {
+  return err
+}
+
+md := rc.GetMetadata()
+md.Labels = map[string]string{"env": "prod"}
+rc.SetMetadata(md)
+
+if err = client.UpdateRemoteCluster(ctx, rc); err != nil {
+  return err
+}
+
+```
+
+## Access Requests
+
+Our [Approval Workflow](enterprise/workflow.md) can be used to dynamically approve and deny a user access to local or remote resources in a trusted cluster.
```

Comment:
nit: there's no `docs/4.4/enterprise/worklow.md` but I'm not sure how this would be handled by the doc builder.

Reply:
Think it should be linked to `enterprise/workflow/index.md` which does exist - the `milv` check on the docs will fail otherwise.